fieldservice_route/models/fsm_order.py

```diff
@@ -31,17 +49,19 @@ def _manage_fsm_route(self, vals):
                 if old_route_id and not old_route_id.order_ids:
                     old_route_id.unlink()
             elif not fsm_route and person_id or scheduled_date_start:
-                fsm_route_obj.create({
-                    'person_id': person_id,
-                    'date': scheduled_date_start,
-                    'route_id': rec.fsm_route_id.id,
-                    'order_ids': [(4, rec.id)]
-                })
+                fsm_route_obj.create(rec.prepare_dayroute_values(
+                    person_id, scheduled_date_start
+                ))
                 if old_route_id and not old_route_id.order_ids:
                     old_route_id.unlink()
 
     @api.model
     def create(self, vals):
+        if not vals.get('person_id'):
```

Comment:
```suggestion
        if not vals.get('person_id') and vals.get('location_id'):
```

Reply:
@nikul-serpentcs location_id is a required field...